libpromises/policy.c

```diff
@@ -582,6 +582,15 @@ static bool ConstraintCheckSyntax(const Constraint *constraint, Seq *errors)
 
     /* Check if lvalue is valid for the bundle's specific section. */
     const PromiseTypeSyntax *promise_type_syntax = PromiseTypeSyntaxGet(bundle->type, section->promise_type);
+
+    if(promise_type_syntax == NULL)
+    {
+        SeqAppend(errors,
+                  PolicyErrorNew(POLICY_ELEMENT_TYPE_CONSTRAINT, constraint,
+                                 POLICY_ERROR_CONSTRAINT_TYPE_MISMATCH, section->promise_type));
```

Comment:
I think this will produce a very confusing error message. Please try it yourself.